packages/hdwallet-provider/src/index.ts

```diff
@@ -36,7 +44,10 @@ class HDWalletProvider {
   private walletHdpath: string;
   private wallets: { [address: string]: ethJSWallet };
   private addresses: string[];
-  private chainId?: number | string;
+  private chainId?: number;
+  private chainSettings: any;
```

Comment:
Can't we reuse the type from constructor? Or does this need to be `any`?

Reply:
I don't see why not. Is it worth the trouble of allowing `chainId` as a string?